go/libraries/doltcore/sqle/dtables/status_table.go

```diff
@@ -0,0 +1,250 @@
+	return NewStatusItr(context, &s)
+}
+
+// NewStatusTable creates a StatusTable
+func NewStatusTable(_ *sql.Context, ddb *doltdb.DoltDB, rsr env.RepoStateReader, drw env.DocsReadWriter) sql.Table {
+	return &StatusTable{ddb: ddb, rsr: rsr, drw: drw}
+}
+
+// BranchItr is a sql.RowItr implementation which iterates over each commit as if it's a row in the table.
+type StatusItr struct {
+	tables   []string
+	isStaged []bool
+	statuses []string
+	idx      int
+}
+
+func NewStatusItr(ctx *sql.Context, st *StatusTable) (*StatusItr, error) {
+	ddb := st.ddb
+	rsr := st.rsr
+	drw := st.drw
+
+	stagedTables, unstagedTables, err := diff.GetStagedUnstagedTableDeltas(ctx, ddb, rsr)
+
+	if err != nil {
+		return &StatusItr{}, err
+	}
+
+	stagedDocDiffs, notStagedDocDiffs, err := diff.GetDocDiffs(ctx, ddb, rsr, drw)
+
+	if err != nil {
+		return &StatusItr{}, err
+	}
+
+	workingTblsInConflict, _, _, err := merge.GetTablesInConflict(ctx, ddb, rsr)
+
+	if err != nil {
+		return &StatusItr{}, err
+	}
+
+	workingDocsInConflict, err := merge.GetDocsInConflict(ctx, ddb, rsr, drw)
+
+	if err != nil {
+		return &StatusItr{}, err
+	}
+
+	tLength := len(stagedTables) + len(unstagedTables) + len(stagedDocDiffs.Docs) + len(notStagedDocDiffs.Docs) + len(workingTblsInConflict) + len(workingDocsInConflict.Docs)
+
+	tables := make([]string, tLength)
+	isStaged := make([]bool, tLength)
+	statuses := make([]string, tLength)
+
+	itr := StatusItr{tables: tables, isStaged: isStaged, statuses: statuses, idx: 0}
+
+	idx := handleStagedTables(stagedTables, &itr, 0)
+	idx = handleUnstagedTables(unstagedTables, &itr, idx)
+	idx = handleStagedDocDiffs(stagedDocDiffs, &itr, idx)
+	idx = handleUnstagedDocDiffs(notStagedDocDiffs, &itr, idx)
+	idx = handleWorkingTablesInConflict(workingTblsInConflict, &itr, idx)
+	idx = handleWorkingDocConflicts(workingDocsInConflict, &itr, idx)
+
+	return &itr, nil
+}
+
+var tblDiffTypeToLabel = map[diff.TableDiffType]string{
+	diff.ModifiedTable: "modified",
+	diff.RenamedTable:  "renamed",
+	diff.RemovedTable:  "deleted",
+	diff.AddedTable:    "new table",
+}
+
+func handleStagedTables(staged []diff.TableDelta, itr *StatusItr, idx int) int {
+	for _, td := range staged {
+		itr.isStaged[idx] = true
+		if td.IsAdd() {
+			itr.tables[idx] = td.CurName()
+			itr.statuses[idx] = tblDiffTypeToLabel[diff.AddedTable]
+		} else if td.IsDrop() {
+			itr.tables[idx] = td.CurName()
+			itr.statuses[idx] = tblDiffTypeToLabel[diff.RemovedTable]
+		} else if td.IsRename() {
+			itr.tables[idx] = fmt.Sprintf("%s -> %s", td.FromName, td.ToName)
+			itr.statuses[idx] = tblDiffTypeToLabel[diff.RemovedTable]
+		} else {
+			itr.tables[idx] = td.CurName()
+			itr.statuses[idx] = tblDiffTypeToLabel[diff.ModifiedTable]
+		}
+
+		idx += 1
+	}
+
+	return idx
+}
+
+func handleUnstagedTables(notStaged []diff.TableDelta, itr *StatusItr, idx int) int {
+	for _, td := range notStaged {
```

Comment:
Lots of repeated code with the function above

Reply:
Good catch.